src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReader.cs

```diff
@@ -1005,7 +1005,7 @@ private SoapArrayInfo ParseArrayType(string value)
             int len = charsLength - pos - 2;
             if (len > 0)
             {
-                string lengthString = new string(chars, pos + 1, len);
+                string lengthString = new string(chars.Slice(pos + 1, len));
```

Comment:
This shouldn't create a new string.  It can e.g. be `ReadOnlySpan<char> lengthString = value.AsSpan(pos + 1, len);`.  If an actual string is needed for the ArgumentException below, then it can be ToString'd only in that exceptional case.

Reply:
@stephentoub  Done, just one question : What do you suggest when using int.TryParse(...) methods to do if this returns false? Now I am throwing exception, which is then catched and thrown the same.